src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -543,18 +553,21 @@ class Simplex_tree {
 
  private:
   /** rec_equal: Checks recursively whether or not two simplex trees are equal, using depth first search. */
-  bool rec_equal(Siblings* s1, Siblings* s2) {
+  template<class OtherSimplexTreeOptions>
+  bool rec_equal(Siblings* s1, typename Simplex_tree<OtherSimplexTreeOptions>::Siblings* s2) {
```

Comment:
```suggestion
  template<class OtherSiblings>
  bool rec_equal(Siblings* s1, OtherSiblings* s2) {
```
? Then we can just call `rec_equal(x,y)` without explicitly giving the template parameter.

Reply:
Why make easy when you can do complicated, I guess?
I think, I wanted to ensure that the other siblings are really siblings. But as it is a private method anyway, there are not too many risk to just template everything.